test/unit/math/prim/prob/poisson_log_test.cpp

```diff
@@ -0,0 +1,53 @@
+#include <stan/math/prim/mat.hpp>
+#include <stan/math/prim/scal.hpp>
+#include <test/unit/math/prim/prob/vector_rng_test_helper.hpp>
+#include <test/unit/math/prim/prob/VectorIntRNGTestRig.hpp>
+#include <gtest/gtest.h>
+#include <boost/math/distributions.hpp>
+#include <boost/random/mersenne_twister.hpp>
+#include <limits>
+#include <vector>
+
+class PoissonLogTestRig : public VectorIntRNGTestRig {
+ public:
+  PoissonLogTestRig()
+      : VectorIntRNGTestRig(10000, 10, {0, 1, 2, 3, 4, 5, 6},
+                            {-0.5, 0.0, 0.1, 1.7}, {-2, 0, 1, 2}, {}, {}) {}
+
+  template <typename T1, typename T2, typename T3, typename T_rng>
+  auto generate_samples(const T1& alpha, const T2&, const T3&,
+                        T_rng& rng) const {
+    return stan::math::poisson_log_rng(alpha, rng);
+  }
+
+  template <typename T1>
+  double pmf(int y, T1 alpha, double, double) const {
+    return std::exp(stan::math::poisson_log_lpmf(y, alpha));
+  }
+};
+
+TEST(ProbDistributionsPoissonLog, errorCheck) {
+  check_dist_throws_all_types(PoissonLogTestRig());
+}
+
+TEST(ProbDistributionsPoissonLog, distributionCheck) {
+  check_counts_real(PoissonLogTestRig());
+}
+
+TEST(ProbPoisson, log_matches_lpmf) {
+  int y = 3;
+  double lambda = 2.3;
+
+  EXPECT_FLOAT_EQ((stan::math::poisson_lpmf(y, lambda)),
+                  (stan::math::poisson_log(y, lambda)));
+  EXPECT_FLOAT_EQ((stan::math::poisson_lpmf<true>(y, lambda)),
+                  (stan::math::poisson_log<true>(y, lambda)));
+  EXPECT_FLOAT_EQ((stan::math::poisson_lpmf<false>(y, lambda)),
+                  (stan::math::poisson_log<false>(y, lambda)));
```

Comment:
This block is in the wrong place. It's very confusing, but we have tests for `poisson_log_lpmf` (Poisson with log parametrization) and tests for `poisson_log` as the old alias for `poisson_lpmf`. Since with this patch the convention is to call tests as `distribution_name_test.cpp`, `poisson_log_test.cpp` should be about `poisson_log_lpmf` and family.

So this block should probably go into `poisson_test`, although doing so won't match what's done for other `log_matches_lpmf`-type tests that are in separate files. Let's put it at the very end of that file, similar as to what we'll do for `neg_binomial_2`.

Reply:
Agree. Done.